datasource/read.go

```diff
@@ -37,4 +50,11 @@ type ReadResponse struct {
 	// source. An empty slice indicates a successful operation with no
 	// warnings or errors generated.
 	Diagnostics diag.Diagnostics
+
+	// DeferredResponse indicates that Terraform should defer
+	// importing this resource.
+	//
+	// This field can only be set if
+	// `(datasource.ReadRequest.ReadClientCapabilities).DeferralAllowed` is true.
+	DeferredResponse *DeferredResponse
```

Comment:
Bikeshed: Do we want to name the field `Deferred` (acknowledging that the `Response` naming is already present in the containing type name) or both the name and type `Deferred` here (matching protocol/terraform-plugin-go)? 

Same for resource package.

Reply:
Ah, good call. I think it'd be fine to drop the `Response` on both the type and field name since it's not really adding any additional context.

I will update the SDKv2 PR to also follow this convention 👍🏻